grakn-graph/src/main/java/ai/grakn/graph/internal/ValidateGlobalRules.java

```diff
@@ -285,20 +285,41 @@ private ValidateGlobalRules() {
      * @param rule the rule to be validated
      * @return Error messages if the rule is not a valid Horn clause (in implication form, conjunction in the body, single-atom conjunction in the head)
      */
-    static Set<String> validateRuleIsHornClause(GraknGraph graph, Rule rule){
+    static Set<String> validateRuleIsValidHornClause(GraknGraph graph, Rule rule){
         Set<String> errors = new HashSet<>();
-        if (rule.getWhen().admin().isDisjunction()){
+        if (rule.getWhen().admin().getDisjunctiveNormalForm().getPatterns().size() != 1){
```

Comment:
I kind of preferred `isDisjunction` much cleaner and more readable. Why did you have to change it?